Tests/TuistLoaderIntegrationTests/ProjectDescriptionHelpers/ProjectDescriptionHelpersBuilderIntegrationTests.swift

```diff
@@ -29,6 +29,51 @@ final class ProjectDescriptionHelpersBuilderIntegrationTests: TuistTestCase {
         super.tearDown()
     }
 
+    func test_build_only_once_even_with_different_subjects() async throws {
+        // Given
+        let path = try temporaryPath()
+        let helpersPath = path
+            .appending(try RelativePath(validating: "\(Constants.tuistDirectoryName)/\(Constants.helpersDirectoryName)"))
+        try FileHandler.shared.createFolder(path.appending(component: Constants.tuistDirectoryName))
+        try FileHandler.shared.createFolder(helpersPath)
+        try FileHandler.shared.write(
+            "import Foundation; class Test {}",
+            path: helpersPath.appending(component: "Helper.swift"),
+            atomically: true
+        )
+
+        let projectDescriptionPath = try await resourceLocator.projectDescription()
+        let searchPaths = ProjectDescriptionSearchPaths.paths(for: projectDescriptionPath)
+
+        let fileSystem = MockFileSystem()
+        let hash = try await ProjectDescriptionHelpersHasher().hash(helpersDirectory: path)
+        let moduleCacheDirectory = path.appending(component: hash)
+        let dylibName = "lib\(ProjectDescriptionHelpersBuilder.defaultHelpersName).dylib"
+        let modulePath = moduleCacheDirectory.appending(component: dylibName)
+        fileSystem.existsResults.insert(modulePath)
+        let fileHandler = MockFileHandling()
+
+        // When
+        let paths = try await Array(0 ..< 3).concurrentMap { _ in
+            let subject = ProjectDescriptionHelpersBuilder(
+                cacheDirectory: path,
+                helpersDirectoryLocator: self.helpersDirectoryLocator,
+                fileHandler: fileHandler,
+                fileSystem: fileSystem
+            )
+            return try await subject.build(
+                at: path,
+                projectDescriptionSearchPaths: searchPaths,
+                projectDescriptionHelperPlugins: []
+            )
+        }
+
+        // Then
+        XCTAssertEqual(Set(paths).count, 1)
+        XCTAssertEqual(fileSystem.existsCounter, 3)
+        XCTAssertEqual(fileHandler.createFolderCount, 0) // should not avoid creating a folder
```

Comment:
I'd consider removing the `fileHandler` completely – that's something we will need to do anyway.

Reply:
Hmmm, I understand, the thing is that this is one of the verification of the test. Checking that the fileHandler.createFolder is not called.

Reply:
> Hmmm, I understand, the thing is that this is one of the verification of the test. Checking that the fileHandler.createFolder is not called.

Is that what we should be checking, though? Shouldn't we instead ensure the command to build the helpers is not run multiple times? Whether the folder is created or not feels like an implementation detail

Reply:
OK, I feel you, this should not be part of the integration tests. I can move it to the unit test level, and indeed check for the system call count like you are going in there.

I would add, that I'd still need to mock the fileSystem there such that I can return `true` for exists and early return -  going to work on that now

Reply:
I have changed the behaviour and moved this logic to the unit test.
Let me know if that makes sense

I will start working on adding Mockable to the file system, and replace my mock here

Reply:
> I would add, that I'd still need to mock the fileSystem there such that I can return true for exists and early return - going to work on that now

Alternatively, you can pre-create the directory and mock the hash that determines the location instead. I think that might be better than mocking the `FileSystem`.

Reply:
As always - great feedback - fixed!